app/services/next_course_creation_step_service.rb

```diff
@@ -1,48 +1,64 @@
 class NextCourseCreationStepService
-  def execute(current_step:, current_provider:)
-    case current_step
-    when :level
-      :subjects
-    when :subjects
-      :age_range
-    when :age_range
-      :outcome
-    when :outcome
-      handle_outcome(current_provider)
-    when :fee_or_salary
-      :full_or_part_time
-    when :apprenticeship
-      :full_or_part_time
-    when :full_or_part_time
-      :location
-    when :location
-      handle_location(current_provider)
-    when :accredited_body
-      :entry_requirements
-    when :entry_requirements
-      :applications_open
-    when :applications_open
-      :start_date
-    when :start_date
-      :confirmation
-    end
+  def execute(current_step:, current_provider:, level:)
+    get_next_step(get_workflow(current_provider, level), current_step)
   end
 
-private
+  def get_next_step(steps, current_step)
+    steps[steps.find_index(current_step).next]
+  end
 
-  def handle_outcome(provider)
-    if provider.accredited_body?
-      :apprenticeship
-    else
-      :fee_or_salary
+  def get_workflow(provider, level)
+    if is_further_education?(level)
+      %i[
+        level
+        outcome
+        full_or_part_time
+        location
+        applications_open
+        start_date
+        confirmation
+      ]
+    elsif is_uni_or_scitt?(provider)
+      %i[
+        level
+        subjects
+        age_range
+        outcome
+        apprenticeship
+        full_or_part_time
+        location
+        entry_requirements
+        applications_open
+        start_date
+        confirmation
+      ]
+    elsif is_school_direct?(provider, level)
+      %i[
+        level
+        subjects
+        age_range
+        outcome
+        fee_or_salary
+        full_or_part_time
+        location
+        accredited_body
+        entry_requirements
+        applications_open
+        start_date
+        confirmation
+      ]
     end
   end
 
-  def handle_location(provider)
-    if provider.accredited_body?
-      :entry_requirements
-    else
-      :accredited_body
-    end
+  def is_school_direct?(provider, level)
+    !(is_uni_or_scitt?(provider) || is_further_education?(level))
+  end
+
+  def is_uni_or_scitt?(provider)
+    provider.accredited_body?
+  end
```

Comment:
Is this the right place for this logic? I'd be really keen to keep this in the model so we can hide this rather messy logic, including `is_school_direct` above.

Reply:
This logic seems tied particularly to this next course creation functionality, I don't think we should move it out until there is demand for it. If there was a need to know this information elsewhere I'd prefer to have it in the backend rather than infering it from sparse properties.

Reply:
I'm worried about divergent implementations of these two methods. I think that given that these are hidden away in a service, if anyone has to implement `is_uni_or_scitt?` and `is_school_direct?` logic again, they won't think to look here, which is why I'd rather this be in the model.

Reply:
isn't this really that?

https://github.com/DFE-Digital/manage-courses-backend/blob/085b2fc4a789c1cf75bce898f6cba90f869d0581/app/models/provider.rb#L50-L56